w5_doubly_linked_list/isaac/CircularyDoublyLinkedList.java

```diff
@@ -0,0 +1,137 @@
+package dataStructure.linkedList.doubly;
+
+public class CircularyDoublyLinkedList {
+    private Node head;
+    private Integer size;
+    public CircularyDoublyLinkedList(){
+        this.head = new Node();
+        this.size = 0;
+    }
+
+    public int getSize(){
+        return this.size;
+    }
+
+    public Node getHead(){
+        return this.head;
+    }
+
+    public Node getFirstNode(){
+        return this.head.getNext();
+    }
+
+    public void splice(Node a, Node b, Node x){
+        Node ap, bn, xn;
+
+        if(a.getPrev() == null && a.getNext() == null) {
+            ap = x;
+            bn = x.getNext();
+
+            ap.setNext(a);
+            a.setPrev(ap);
+            bn.setPrev(a);
+            a.setNext(bn);
+        } else {
+            ap = a.getPrev();
+            bn = b.getNext();
+            xn = x.getNext();
+
+            ap.setNext(bn);
+            bn.setPrev(ap);
+            x.setNext(a);
+            a.setPrev(x);
+            xn.setPrev(b);
+            b.setNext(xn);
+        }
+    }
+
+    // a를 x노드 뒤로 이동한다.
+    public void moveAfter(Node a, Node x){
+        splice(a, a, x);
+    }
+
+    public void moveBefore(Node a, Node x){
+        splice(a, a, x.getPrev());
+    }
+
+    public void insertAfter(Node x, String key){
+        moveAfter(new Node(key), x);
+        this.size++;
+    }
+
+    public void insertBefore(Node x, String key){
+        moveBefore(new Node(key), x);
+        this.size++;
+    }
+
+    public void pushFront(String key){
+        insertBefore(this.head.getNext(), key);
+    }
+
+    public void pushBack(String key){
+        if(this.head.getNext() == null) {
+            pushFront(key);
+        }
+        else {
+            Node lastNode = this.head;
+            while (lastNode.getNext().getKey() != null) {
+                lastNode = lastNode.getNext();
+            }
+            insertAfter(lastNode, key);
```

Comment:
여기도 `insertBefore`를 이용해서 처리하면 `while`문을 제거할 수 있을 것 같아요!
```suggestion
            insertBefore(this.head, key);
```